cli/src/commands/extensions/permissions.rs

```diff
@@ -61,6 +62,74 @@ impl Permission {
             Err(anyhow!("Requires {resource_type} access to {resource:?}"))
         }
     }
+
+    pub fn subset_of(&self, parent: &Permission) -> Result<Permission> {
+        use Permission::*;
+        match (parent, self) {
+            // Child deny-all always succeeds, returning deny-all.
+            (_, &Boolean(false)) => Ok(Boolean(false)),
+            // Parent deny-all fails with all child permissions but deny-all.
+            (&Boolean(false), _) => Err(anyhow!("Requested permissions incompatible with parent")),
+            // Parent allow-all always succeeds, returning the child's permissions.
+            (&Boolean(true), child) => Ok(child.clone()),
+            // Child allow-all fails with more restrictive parent permissions.
+            (_, &Boolean(true)) => Err(anyhow!("Requested permissions incompatible with parent")),
+            // Parent set vs child set have to be validated.
+            // This will error if child is not subset of parent, and return the child set otherwise.
+            (&List(ref parent), &List(ref child)) => Permission::paths_subset(parent, child)
+                .map(|()| Permission::List(child.clone()))
+                .map_err(|mismatches| {
+                    anyhow!(
+                        "Requested permission paths incompatible with parent: {}",
+                        mismatches.join(", ")
```

Comment:
```
Requested permission paths incompatible with the manifest: /tmp/foo, /dev
```

Reply:
That's probably going to be a bit difficult to connect back with `runSanboxed`, I'd love to be able to tell the extension author that's where the error came from.

I also think that printing this error without mentioning the specific permission is problematic. Imagine a user accidentally swapping read/write, it would likely be much faster to troubleshoot when we print the specific permission in the output here.

Reply:
We deal with that [here](https://github.com/phylum-dev/cli/pull/746/files#diff-061d5eeda494ae877735811ed5f387c830d778a7f49a345622fc17ba8f17f0f6R228-R235).

The overall output should be something like
```
read: Requested permissions incompatible with the manifest: /here/there, /tmp/something
```
Not married to the style, so I'm open to suggestions here.

Reply:
I like that a lot better. I also updated the other error messages to state "permissions are denied by the manifest".